TigranKirakosov/index.js

```diff
@@ -0,0 +1,253 @@
+      };
+    },
+    getDOMStrings: function() {
+      return DOMStrings;
+    },
+    deleteListItem: function(selectorID) {
+      var element;
+      element = document.querySelector(`#${selectorID}`);
+      element.parentNode.removeChild(element);
+    },
+    toggleTask: function(icon) {
+      if (icon.className === DOMStrings.checkedBtn) {
+        icon.className = DOMStrings.uncheckedBtn;
+      } else if (icon.className === DOMStrings.uncheckedBtn) {
+        icon.className = DOMStrings.checkedBtn;
+      }
+    },
+    filterTasks: function(tasks) {
+      var filterValue, filteredTasks, taskList;
+      var html = '';
+      filterValue = document.querySelector(DOMStrings.filterInput).value;
+      taskList = document.querySelector(DOMStrings.taskList);
+
+      filteredTasks = tasks.filter(function(task) {
+        return task.deadline == filterValue || task.isDone == filterValue;
+      });
+      filteredTasks.forEach(function(task) {
+        html += `
+        <li class="collection__item" id="item-${task.id}">
+          <span class="text">${task.text}</span>
+          <div class="icons">
+            <i class="${task.iconClass}"></i>
+            <i class="fas fa-backspace"></i>
+          </div>
+          <span class="deadline">Deadline: ${task.deadline}</span>
+        </li>
+        `;
+      });
+      taskList.innerHTML = html;
+    },
+    clearTasks: function() {
+      document.querySelector(DOMStrings.taskList).innerHTML = '';
+    },
+    clearFields: function() {
+      document.querySelector(DOMStrings.taskInput).value = '';
+      document.querySelector(DOMStrings.deadlineInput).value = 'no deadline';
+      document.querySelector(DOMStrings.taskInput).focus();
+    }
+  };
+})();
+
+var storageController = (function(UICtrl) {
+  var Task, tasks, DOM;
+  DOM = UICtrl.getDOMStrings();
+
+  Task = function(id, text, deadline) {
+    this.id = id;
+    this.text = text;
+    this.deadline = deadline;
+    this.isDone = 'undone';
+    this.iconClass = DOM.uncheckedBtn;
+  };
+  Task.prototype.toggleState = function() {
+    if (this.isDone === 'undone') {
+      this.isDone = 'done';
+      this.iconClass = DOM.checkedBtn;
+    } else if (this.isDone === 'done') {
+      this.isDone = 'undone';
+      this.iconClass = DOM.uncheckedBtn;
+    }
+  };
+  if (
+    localStorage.getItem('tasks') === null ||
+    localStorage.getItem('tasks') === ''
+  ) {
+    tasks = [];
+  } else {
+    tasks = JSON.parse(localStorage.getItem('tasks'));
+  }
+  return {
+    getFromLS: function() {
+      return tasks;
+    },
+    addTask: function(text, deadline) {
+      var newTask, ID;
+      if (tasks.length > 0) ID = tasks[tasks.length - 1].id + 1;
+      else ID = 0;
+      newTask = new Task(ID, text, deadline);
+      return newTask;
+    },
+    addToLS: function(task) {
+      tasks.push(task);
+      localStorage.setItem('tasks', JSON.stringify(tasks));
+    },
+    deleteFromLS: function(id) {
+      var ids, index;

```

Comment:
again - I don't think `deleteFromLS` is good naming for this function - you delete task from internal state, and just update `localStorage` value